### PR TITLE
feat: add action to publish db docs to gh-pages

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

This action will run on every commit to `main`, rather than requiring any human intervention/possibility of forgetting to update the db.

Related to https://github.com/freeCodeCamp/chapter/pull/639 in that it introduces the same `db:sync` command.

### CODE REVIEW

**Comment by allella** [comment]
@ojeytonwilliams is there anything holding this PR back?

I had a note to update the docs and old issues to specify that this is automated.

**Comment by ojeytonwilliams** [comment]
> is there anything holding this PR back?

Not a good reason, no.  This uses [typeorm schema:sync](https://github.com/typeorm/typeorm/blob/master/docs/using-cli.md#sync-database-schema) to make sure the db schema is correct before building the diagram.  I was still wondering if we should use that and do away with the migrations, as discussed here: https://github.com/freeCodeCamp/chapter/pull/639

I've had time to think it over and I'm pretty confident the reduction in developer friction of automatic db syncing is worth it.  We should revisit migrations once we're happy with the MVP, but until then I'd rather nobody had to worry about them.

TL;DR: nope, merge away whenever you're ready.

**Comment by allella** [comment]
@ojeytonwilliams  
The [schema seems to have updated on the website](https://opensource.freecodecamp.org/chapter/), as indicated by the time stamp, but I did get an [CI error email that goes to this page](https://github.com/allella/chapter/actions/runs/1258623952).


image


image

**Comment by ojeytonwilliams** [comment]
@allella it will be failing because actions on forks don't have write permission.  I'll restrict the action to this repo to stop the spam.

**Comment by allella** [comment]
This PR is merged and now updates to the _gh-pages_ branch and [online schema](https://opensource.freecodecamp.org/chapter/) are now automatically triggered by commits to the _main_ branch, so we no longer require [manual steps](https://github.com/freeCodeCamp/chapter/issues/54#issuecomment-799653569).

Posting this for future reference in case someone needs to tweak things or wants to know how to work with SchemaSpy locally or on a different project.
